### PR TITLE
#9 Some minor refactor work

### PR DESCRIPTION
**Pull request by sleepylee**
## What happened 👀

Just some minor refactor work after discussion in the previous PR.
 
## Insight 📝
- Change `async/await` to functional chains `then()/onError()` because I like functional programming 😎 
- Make Result constructor private; to emphasize the need of using only Success/Failed as the return type.
 
## Proof Of Work 📹
The app works like before

### CODE REVIEW

File: lib/usecases/login_use_case.dart [hunk 1]
```diff
@@ -21,14 +22,11 @@ class LoginUseCase extends UseCase<AuthToken, LoginCredential> {
...
-    } catch (e) {
-      final apiError = ApiError(NetworkExceptions.getDioException(e), e);
-      return Future.value(Failed(apiError));
-    }
+  Future<Result<AuthToken>> call(LoginCredential credential) {
+    return _repository
+        .login((credential.email), credential.password)
+        .then<Result<AuthToken>>((value) => Success(value))
+        .onError<DioError>((error, stackTrace) => Failed(
+            UseCaseError(NetworkExceptions.getDioException(error), error)));
```
Comment (markgravity) [inline lib/usecases/login_use_case.dart:30]: I suggest renaming 
`NetworkExceptions.getDioException(error), error)` -> `NetworkExceptions.fromDioException(error), error)`
will describe more correctly what this function will do 👍

File: lib/usecases/use_case_result.dart [hunk 2]
```diff
@@ -1,22 +1,24 @@
...
 
 class Success<T> extends Result<T> {
   final T value;
 
-  Success(this.value);
+  Success(this.value) : super._();
 }
 
-class ApiError implements Exception {
+class UseCaseError implements Exception {
```
Comment (markgravity) [inline lib/usecases/use_case_result.dart:13]: What do you think about rename `UserCaseError` to `UserCaseException`?
Refer from [Error](https://api.dart.dev/stable/2.12.2/dart-core/Error-class.html) & [Exception](https://api.dart.dev/stable/2.12.2/dart-core/Exception-class.html)

Comment (sleepylee) [reply lib/usecases/use_case_result.dart:13]: for this renaming it makes sense. Subclass must extends the superclass name 🤙

File: lib/usecases/login_use_case.dart (hunk 1 above)
Comment (markgravity) [reply lib/usecases/login_use_case.dart]: how about this @sleepylee 😁

File: lib/use_cases/login_use_case.dart [hunk 3]
```diff
@@ -21,14 +21,11 @@ class LoginUseCase extends UseCase<AuthToken, LoginCredential> {
...
-    } catch (e) {
-      final apiError = ApiError(NetworkExceptions.getDioException(e), e);
-      return Future.value(Failed(apiError));
-    }
+  Future<Result<AuthToken>> call(LoginCredential credential) {
+    return _repository
+        .login((credential.email), credential.password)
+        .then<Result<AuthToken>>((value) => Success(value))
+        .onError((err, stackTrace) => Failed(
+            UseCaseException(NetworkExceptions.getDioException(err), err)));
```
Comment (sleepylee) [inline lib/use_cases/login_use_case.dart:29]: @markgravity lol I updated it; I think you have to review this latest version 🤪

File: lib/usecases/login_use_case.dart (hunk 1 above)
Comment (sleepylee) [reply lib/usecases/login_use_case.dart]: ohhh `get` -> `from` 🤔  not a bad idea, let me update it 🙆‍♂️

**Review by manh-t — APPROVED** [review]
sorry for late 💯

**Pull request merged**